### ISSUE TITLE
SDP answer fingerprint algorithm should be case-insentitive?

### ISSUE THREAD

**Issue by achingbrain** [paullouisageneau/libdatachannel#968]
I'm trying to set a remote description on a peer connection, the fingerprint is in the SDP handshake as:

```
a=fingerprint:SHA-256 {fingerpint}
```

It causes libdatachannel to fail with a message "Remote description has no valid fingerprint" but it works in Chrome.

Changing it to the following makes it work with libdatachannel and chrome:

```
a=fingerprint:sha-256 {fingerpint}
```

The [example SDP answers](https://datatracker.ietf.org/doc/html/rfc8841#section-13) in the spec have `a=fingerprint:SHA-256 {fingerprint}` so libdatachannel should accept upper case here.

Granted now there is code out there that will supply lower-case fingerprint algorithm names so maybe libdatachannel should be case-insensitive the same as Chrome to avoid breaking any existing usage?

**Comment by paullouisageneau** [comment]
According to [RFC 4572 section 5](https://www.rfc-editor.org/rfc/rfc4572#section-5) the hash function name must be lowercase:
```
hash-func              =  "sha-1" / "sha-224" / "sha-256" /
                             "sha-384" / "sha-512" /
                             "md5" / "md2" / token
                             ; Additional hash functions can only come
                             ; from updates to [RFC 3279](https://www.rfc-editor.org/rfc/rfc3279)
```

However the [examples in RFC 8841](https://datatracker.ietf.org/doc/html/rfc8841#section-13) indeed use lower case, and it looks like other implementations accept both, so libdatachannel should do the same.

**Comment by paullouisageneau** [comment]
Released in 0.19.1

**Comment by achingbrain** [comment]
Amazing, thanks for the speedy resolution!
